pennylane/wires.py

```diff
@@ -493,6 +493,207 @@ def unique_wires(list_of_wires):
 
         return Wires(tuple(unique), _override=True)
 
+    def union(self, other):
+        """Return the union of the current Wires object and either another Wires object or an
+        iterable that can be interpreted like a Wires object e.g. List.
+
+        Args:
+            other (Wires): Another Wires object to perform the union with.
```

Comment:
The doc above says it can be a list as well, please update here and everywhere.

[optional] consider removing `Another`.

Reply:
It's a bit more complex than that I realized. So I changed it to this
```
other (Any): Wires or any iterable that can be interpreted like a Wires object 
    to perform the union with. See _process for details on the interpretation. 
```